ros2topic/ros2topic/verb/pub.py

```diff
@@ -82,6 +82,10 @@ def add_arguments(self, parser, cli_name):
             help=(
                 'Wait until finding the specified number of matching subscriptions. '
                 'Defaults to 1 when using "-1"/"--once"/"--times", otherwise defaults to 0.'))
+        parser.add_argument(
+            '--max-wait-time', type=positive_float, default=None,
+            help=(
+                'This sets the maximum wait time if --wait-until-matching-subscriptions is set'))
```

Comment:
I think we should be clear about the units here.  Presumably this is in seconds, so at the very least we should say that in the help.  We also might consider making the argument `--max-wait-time-sec`, to make it really clear.  Same goes for the variable names below.

Reply:
besides, probably add default behavior explanation, that is wait forever until it finds expected matching subscription.

Reply:
See c6dda6e